ts/src/bybit.ts

```diff
@@ -6037,348 +4557,135 @@ export default class bybit extends Exchange {
-        }
-    }
-
-    async fetchOrderTrades (id: string, symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
-        /**
-         * @method
-         * @name bybit#fetchOrderTrades
-         * @description fetch all the trades made from a single order
-         * @param {string} id order id
-         * @param {string} symbol unified market symbol
-         * @param {int} [since] the earliest time in ms to fetch trades for
-         * @param {int} [limit] the maximum number of trades to retrieve
-         * @param {object} [params] extra parameters specific to the bybit api endpoint
-         * @returns {object[]} a list of [trade structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#trade-structure}
-         *
-         */
+        const isUnifiedAccount = (enableUnifiedMargin || enableUnifiedAccount);
         const request = {};
-        const clientOrderId = this.safeString2 (params, 'clientOrderId', 'orderLinkId');
-        if (clientOrderId !== undefined) {
-            request['orderLinkId'] = clientOrderId;
-        } else {
-            request['orderId'] = id;
-        }
-        params = this.omit (params, [ 'clientOrderId', 'orderLinkId' ]);
-        return await this.fetchMyTrades (symbol, since, limit, this.extend (request, params));
-    }
-
-    async fetchMyUnifiedTrades (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
-        await this.loadMarkets ();
         let market = undefined;
-        const request = {
-            // 'symbol': market['id'],
-            // 'category': '', // Product type. spot,linear,option
-            // 'orderId': '', // Order ID
-            // 'orderLinkId': '', // User customised order ID
-            // 'baseCoin': '', // Base coin
-            // 'startTime': 0, // The start timestamp (ms)
-            // 'endTime': 0, // The end timestamp (ms)
-            // 'execType': '', // Execution type
-            // 'limit': 0, // Limit for data size per page. [1, 100]. Default: 50
-            // 'cursor': '', // Cursor. Used for pagination
-        };
+        let type = this.safeString (params, 'type', 'linear');
+        let isUsdcSettled = false;
         if (symbol !== undefined) {
             market = this.market (symbol);
+            isUsdcSettled = market['settle'] === 'USDC';
             request['symbol'] = market['id'];
-        }
-        let type = undefined;
-        [ type, params ] = this.handleMarketTypeAndParams ('fetchMyTrades', market, params);
-        if (type === 'spot') {
-            request['category'] = 'spot';
+            if (market['spot']) {
+                type = 'spot';
+            } else if (market['linear']) {
+                type = 'linear';
+            } else if (market['inverse']) {
+                type = 'inverse';
+            } else if (market['option']) {
+                type = 'option';
```

Comment:
@sc0Vu we have this logic everywhere, so maybe we can create a helper method like `getBybitType(name,market, params)` , example

```Javascript
    getBybitType (method, market, params) {
        let type = undefined;
        [ type, params ] = this.handleMarketTypeAndParams (method, market, params);
        let subType = undefined;
        [ subType, params ] = this.handleSubTypeAndParams (method, market, params);
        if (type === 'option' || type === 'spot') {
            return type;
        }
        return subType;
    }
```